### PR TITLE
Issue-543 JdbcDatabaseContainer now accepts Future as image name

### LINKED ISSUE #543: JdbcDatabaseContainer doesn't have lazy image constructor
Hi

Is there a reason why JdbcDatabaseContainer doesn't have the super constructor with Future<String> image)?

```
public JdbcDatabaseContainer(@NonNull final Future<String> image) {
        super(image);
}
```

### PR DESCRIPTION
**Pull request by iNikem**
Fixes #543 

But now I have a question: may be a better approach would be to accept `Supplier` instead of `Future`? It seems to suit the desired functionality better.

### CODE REVIEW

**Comment by iNikem** [comment]
@bsideup Any idea why gradle fails on Travis? Works on my machine :)

**Comment by AnkBurov** [comment]
@iNikem I think you accidentally changed gradle-wrapper.jar and added it to commit. Try revert  all changes with gradle-wrapper.jar
>$ ./gradlew build -x check
Error: Could not find or load main class org.gradle.wrapper.GradleWrapperMain

https://github.com/testcontainers/testcontainers-java/pull/582/files#diff-887141a5ba95530593029fed3eef7684

**Comment by bsideup** [comment]
@kiview

**Comment by bsideup** [comment]
@iNikem we're fixing the issue with corrupted JAR files in the master branch, could you please revert your Gradle Wrapper change and keep only code change for now? :)

**Comment by iNikem** [comment]
@bsideup I have reverted it to your commit, but it still shows as changed.

**Comment by bsideup** [comment]
@iNikem yes, because of incorrect `.gitattributes`.
You can do:
```shell
git reset master
git add modules/jdbc/src/main/java/org/testcontainers/containers/JdbcDatabaseContainer.java
git commit -m "Closes #543 JdbcDatabaseContainer now accepts Future as image name"
```

**Comment by iNikem** [comment]
@bsideup Ok, Travis is green. Approve and merge? :)

File: modules/jdbc/src/main/java/org/testcontainers/containers/JdbcDatabaseContainer.java [hunk 1]
```diff
@@ -140,6 +154,20 @@ public Connection createConnection(String queryString) throws SQLException {
...
+     * Template method for constructing the JDBC URL to be used for creating {@link Connection}s.
+     * This should be overridden if the JDBC URL and query string concatenation or URL string
+     * construction needs to be different to normal.
+     *
+     * @param queryString
+     *          query string parameters that should be appended to the JDBC connection URL.
+     *          The '?' character must be included
+     * @return a full JDBC URL including queryString
+     */
+    protected String constructUrlForConnection(String queryString) {
```
Comment (bsideup) [inline modules/jdbc/src/main/java/org/testcontainers/containers/JdbcDatabaseContainer.java]: This change is unrelated, isn't it?

**Comment by bsideup** [comment]
@iNikem please mention your change in `CHANGELOG.md`

File: modules/jdbc/src/main/java/org/testcontainers/containers/JdbcDatabaseContainer.java (hunk 1 above)
Comment (iNikem) [reply modules/jdbc/src/main/java/org/testcontainers/containers/JdbcDatabaseContainer.java]: No, and I haven't done them. Will try to revert...

**Comment by iNikem** [comment]
Now I am quite confused about what is going on with this PR and how changes appear here. Let me decline it and make a fresh new branch for this.

**Comment by kiview** [comment]
Yes sorry about the `.gitattribute` problems, seems the impact was higher than expected 😅
